### PR TITLE
VOLDEV-181 Update CheckUser "Replace talk pages" function to work with Message Wall

### PR DESCRIPTION
**Pull request by ty-a**

### CODE REVIEW

**Comment by mszabo-wikia** [comment]
CR handled in https://wikia-inc.atlassian.net/browse/SUS-1520

File: extensions/CheckUser/CheckUser.i18n.php [hunk 1]
```diff
@@ -52,6 +52,7 @@
 IP addresses will be blocked for one week for anonymous users only and account creation will be disabled.',
 	'checkuser-blocktag' => 'Replace user pages with:',
 	'checkuser-blocktag-talk' => 'Replace talk pages with:',
+	'checkuser-blocktag-wall' => 'Replace Message Wall Greeting with:',
```
Comment (mixth-sense) [inline extensions/CheckUser/CheckUser.i18n.php]: :white_check_mark:

File: extensions/CheckUser/CheckUser_body.php [hunk 2]
```diff
@@ -38,7 +38,7 @@ public function execute( $subpage ) {
 		$talkTag = $wgRequest->getBool( 'usettag' ) ?
 			trim( $wgRequest->getVal( 'talktag' ) ) : '';
 
-		$m = array();
+		$m = [];
```
Comment (mixth-sense) [inline extensions/CheckUser/CheckUser_body.php]: :white_check_mark:

File: extensions/CheckUser/CheckUser_body.php [hunk 3]
```diff
@@ -92,8 +92,8 @@ public function execute( $subpage ) {
 	protected function preCacheMessages() {
 		// Precache various messages
 		if ( !isset( $this->message ) ) {
-			foreach ( array( 'diff', 'hist', 'minoreditletter', 'newpageletter', 'blocklink', 'log' ) as $msg ) {
-				$this->message[$msg] = wfMsgExt( $msg, array( 'escape' ) );
+			foreach ( [ 'diff', 'hist', 'minoreditletter', 'newpageletter', 'blocklink', 'log' ] as $msg ) {
+				$this->message[$msg] = wfMessage( $msg )->escaped();
```
Comment (mixth-sense) [inline extensions/CheckUser/CheckUser_body.php]: :white_check_mark:

File: extensions/CheckUser/CheckUser_body.php [hunk 4]
```diff
@@ -110,9 +110,9 @@ public function getCheckUserLogTitle() {
 
 	protected function showGuide() {
 		global $wgOut;
-		$wgOut->addWikiText( wfMsg( 'checkuser-summary' ) .
+		$wgOut->addWikiText( wfMessage( 'checkuser-summary' )->text() .
 			"\n\n[[" . $this->getCheckUserLogTitle()->getPrefixedText() .
-				'|' . wfMsg( 'checkuser-showlog' ) . ']]'
+				'|' . wfMessage( 'checkuser-showlog' )->text() . ']]'
```
Comment (mixth-sense) [inline extensions/CheckUser/CheckUser_body.php]: :white_check_mark:

File: extensions/CheckUser/CheckUser_body.php [hunk 5]
```diff
@@ -319,22 +326,22 @@ protected function noMatchesMessage( $userName, $checkLast = true ) {
 			if ( $user_id ) {
 				$revEdit = $dbr->selectField( 'revision',
```
Comment (ghost) [inline extensions/CheckUser/CheckUser_body.php:327]: Does `selectField()` imply `LIMIT 1`? Considering `$lastEdit = max( $revEdit, $logEdit );` one is all we need.

**Review by ghost — DISMISSED** [review]

**Review by ghost — CHANGES_REQUESTED** [review]
Please, address the `LIMIT 1` comment, I have made.

File: extensions/CheckUser/CheckUser_body.php (hunk 5 above)
Comment (ty-a) [reply extensions/CheckUser/CheckUser_body.php:327]: According to https://github.com/Wikia/app/blob/dev/includes/db/Database.php#L1201, it does.

Comment (ghost) [reply extensions/CheckUser/CheckUser_body.php:327]: Cool. Then it does not need to be explicitly stated here. Thank you very much for double-checking.

**Review by ghost — APPROVED** [review]

**Comment by ghost** [comment]
@ty-a ― good job, thank you very much for your contribution!

This PR will be merged as soon as it is QA'd.

**Pull request merged**